pkg/auth/providers/oidc/oidc_client.go

```diff
@@ -57,3 +58,36 @@ func AddCertKeyToContext(ctx context.Context, certificate, key string) (context.
 
 	return oidc.ClientContext(ctx, client), nil
 }
+
+func FetchAuthURL(config map[string]interface{}) (string, error) {
+	// If the authEndpoint is already configured, use that
+	if authURL, ok := config["authEndpoint"]; ok {
+		return authURL.(string), nil
+	}
+
+	issuerURL, ok := config["issuer"].(string)
+	if !ok {
+		return "", fmt.Errorf("both authEndpoint and issuerURL are missing in the authConfig")
+	}
```

Comment:
Should we apply the same logic to how we access and type-assert these two values?

Reply:
yes!